joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/base/FileCreationPass.scala

```diff
@@ -38,21 +40,25 @@ class FileCreationPass(cpg: Cpg) extends CpgPass(cpg) with LinkingUtil {
         dstGraph.addEdge(srcNode, newFile, EdgeTypes.SOURCE_FILE)
       }
     }
-
-    // Create SOURCE_FILE edges from nodes of various types to FILE
-    linkToSingle(
-      cpg,
-      srcNodes = cpg.graph.nodes(srcLabels*).toList,
-      srcLabels = srcLabels,
-      dstNodeLabel = NodeTypes.FILE,
-      edgeType = EdgeTypes.SOURCE_FILE,
-      dstNodeMap = { x =>
-        originalFileNameToNode.get(x)
-      },
-      dstFullNameKey = PropertyNames.FILENAME,
-      dstGraph,
-      Some(createFileIfDoesNotExist)
-    )
+    try {
+      // Create SOURCE_FILE edges from nodes of various types to FILE
+      linkToSingle(
```

Comment:
also add handling for method `linkToSingle`

Reply:
done